src/range.rs

```diff
@@ -183,34 +185,38 @@ impl<V: Version> Range<V> {
                 }
 
                 // Right contains an infinite interval:
-                (Some((l1, Some(l2))), Some((r1, None))) => {
-                    if l2 < r1 {
+                (Some((l1, Some(l2))), Some((r1, None))) => match l2.cmp(r1) {
+                    Ordering::Less => {
```

Comment:
This feels a lot less readable to me. What is Clippy saying about this one?

Reply:
warning: `if` chain can be rewritten with `match`
   --> src/range.rs:202:21
    |
202 | /                     if r2 < l1 {
203 | |                         right = right_iter.next();
204 | |                     } else if r2 == l1 {
205 | |                         segments.extend(right_iter.cloned());
...   |
211 | |                         break;
212 | |                     }
    | |_____________________^
    |
    = help: Consider rewriting the `if` chain to use `cmp` and `match`.
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain

Reply:
I like it because it shows the intent, you are performing a comparison over all possible values.

There's a principle I use while coding that can be formulated as "use the least powerful construct that does that job".

Like when you are iterating over values in a list, `foreach` is preferred over `for (int i = 0; i < len; ++i)` since the latter is more powerful construct, using that indicates that you probably want to manipulate indexes somehow. If you aren't doing that, `foreach` says "simple iteration".

Same deal here: match over possible comparison values says exactly that. If chain is more powerful, you would use it if you wanted to do something additional in the conditions and you have to check whether that is the case. It is not the case here, hence we can use less powerful tool and save ourselves condition checking.